src/main/java/org/ballerinax/datamapper/DataMapperNodeVisitor.java

```diff
@@ -0,0 +1,208 @@
+                }
+            }
+        }
+    }
+
+    @Override
+    public void visit(ClassDefinitionNode classDefinitionNode) {
+        NodeList<Token> classTypeQualifiers = classDefinitionNode.classTypeQualifiers();
+        for (Token classTypeQualifier : classTypeQualifiers) {
+            if (classTypeQualifier.text().equals("client")) {
+                if (model.symbol(classDefinitionNode).isPresent()) {
+                    Symbol symbol = model.symbol(classDefinitionNode).get();
+                    if (symbol.getName().isEmpty()) {
+                        continue;
+                    }
+                    String clientName = symbol.getName().get();
+                    Collection<MethodSymbol> methods = ((BallerinaClassSymbol) symbol).methods().values();
+                    Map<String, FunctionRecord> functionMap = new HashMap<>();
+                    for (MethodSymbol method : methods) {
+                        boolean checkForRemote = false;
+                        for (Qualifier qualifier : method.qualifiers()) {
+                            if (qualifier == Qualifier.REMOTE) {
+                                checkForRemote = true;
+                                break;
+                            }
+                        }
+                        if (checkForRemote) {
+                            if (method.getName().isEmpty()) {
+                                continue;
+                            }
+                            String functionName = method.getName().get();
+                            FunctionRecord functionRecord = new FunctionRecord();
+                            for (ParameterSymbol parameter : method.typeDescriptor().parameters()) {
+                                if (parameter.getName().isPresent()) {
+                                    String parameterName = parameter.getName().get();
+                                    String paraType = parameter.typeDescriptor().typeKind().toString();
+                                    functionRecord.addParameter(parameterName, paraType);
+                                }
+                            }
+                            TypeSymbol returnTypeSymbol = method.typeDescriptor().returnTypeDescriptor().get();
+                            String returnName = null;
+                            if (returnTypeSymbol.typeKind() == TypeDescKind.UNION) {
+                                List<TypeSymbol> returnList = ((UnionTypeSymbol) returnTypeSymbol).
+                                        memberTypeDescriptors();
+                                for (TypeSymbol typeSymbol : returnList) {
+                                    if (typeSymbol.typeKind() == TypeDescKind.ERROR) {
+                                        continue;
+                                    } else {
+                                        returnName = typeSymbol.signature();
+                                        break;
```

Comment:
Why break it here? Say the return type is `error|T1|T2|T3`. Then `returnName` will be just `T1`.

Reply:
Modified the code according to the comment